File: BeA.java
public class BeA{
	public static volatile boolean running = true;

	public static void main(String[] args){
		Runtime.getRuntime().addShutdownHook(new Thread(){
			public void run(){
				BeA.running = false;
			}
		});
		while(BeA.running){
			System.out.println("NOW'S YOUR CHANCE TO BE A");
		}
		System.out.println("BIG SHOT");
	}

}

File: bea.c
#include <stdio.h>
#include <stdbool.h>
#include <signal.h>


static volatile bool running = true;

void handle_sigint(int sig){
	running = false;
}

int main(){
	signal(SIGINT, handle_sigint);
	while(running){
		printf("NOW'S YOUR CHANCE TO BE A\n");
	}
	printf("BIG SHOT\n");
}
